### ISSUE TITLE
optifabric and fabric api compatibility issue: 100% map load then stuck

### ISSUE THREAD

**Issue by Pixie-Kitty** [Chocohead/OptiFabric#726]
TLDR at the bottom. This is long, but it explains the steps and testing I did to find this out. I'm not coder/programmer or into mods all that much, just a noob trying to add cool things to my game. And trying to help out those who might not fully understand what's going on with basic understanding I learned.

My game loads fine with Fabric and Optifabric. However, when loading a world, the loading get stuck at 100%. I've cracked it down to being an issue with compatibility with the newest versions of Fabric API and Optifabric. When removing Optifabric from my mod's folder, I can load the game fine with fabric, the world loading like normal. However, I can't use shaders or some resource packs that rely on optifine. If I want to use mods that use fabric I won't be able to use anything that relies on optifine without optifabric, but it's not working with fabric API, so I can only use one or the other right now. Either no mods and only resource packs, shaders, or no resource packs or shaders and only mods.

 So I tested a few things to see what the problem was, here's what I found out:

I can't load the game at all using fabric without Fabric API and get an error to install Fabric even tho it is installed just without the API in my mod's folder. So fabric API needs to be in my mods folder otherwise game won't load at all. I believe I loaded the game completely fine without the API at one point, but can't be sure, so this is why I believe it to be part of the problem, but my memory could be wrong, and I'd always needed fabric API. Still not sure, tho, it has been a stressful and tiring couple of days. Fabric, fabric API, and optifabric used to work fine before I updated to 1.18.2  

If I removed Optifabric my resource packs and shaders don't work while using fabric, (kind of given but thought I'd mention it) making the textures like the better leaves pack look like black lines because some packs rely on optifine. If I just use Optifine the whole game is fine, I just can't use mods, hence having optifabric for use of all packs, shaders, and mods. Other packs like the Mizuno's 16 packs works completely fine using fabric when not having optifabric also works fine just with optifine.

 I'm super new to all this modded stuff, so bare with me. This is just what I learned so far over hours and days trying to get it to work, hopefully it's helpful. It's all so confusing to me, and I'm trying to learn, but so far this is what I've come to understand.
 
The steps I took:

~ I figured out the problem was the Optifabric and Fabric API compatibility because when I removed optifabric from the mod's folder the game worked fine, not causing it to get stuck at 100% map loading. I tested it while removing Fabric API and keeping optifabric and the game worked fine with no issues. The next time I loaded the game it gave me an error that I needed fabric even tho I have the current version of fabric, so I reinstalled fabric and that didn't work game gave me the same error and couldn't load the game at all. I figured it was because I didn't have the API.

 I input the fabric API back to my mod's folder, tried it again with optifabric game loaded fine, but it couldn't load the world, getting stuck at 100% with a java runtime error. Same error I've been trying to figure out a way around. So I ran it all again without optifabric it loaded fine without the world being stuck at 100% no errors, however because I didn't have optifabric I wasn't able to use shaders or some resource packs.

 From my basic mods understanding fabric API is definitely needed to use fabric at all which was weird because I used it without it before. Though I could be wrong, my memory fails me sometimes. I've tested other mods I have, and it comes down to these two only. Seems like many others are having this issue, as I've been dealing with it for days after updating my game to 1.18.2. I tried moving over to sodium but some packs I use work with optifine only, and I would like to have mods still while using those packs, so I hope this issue can get solved and hope I helped in my round about way of explaining and being the noob I am at all this confusing modding stuff. I am assuming the problem is optifabric. If it's not, then I have no other ideas of what it could be.

TLDR: Optifabric and fabric API aren't compatible by reasons of optifabric causing the game to get stuck upon 100% loading a world map, causing the game to crash. Removing optifabric from the mod's folder resolves this issue, however anything used that relies on optifabric for optifine like shaders and resource packs can not be used.

**Comment by tyukara** [comment]
I had the same problem. fabric api 0.53.4+1.18.2 is the cause. fabric api 0.53.0+1.18.2 will work around the problem.

**Comment by ChromaticIsobar** [comment]
> I had the same problem. fabric api 0.53.4+1.18.2 is the cause. fabric api 0.53.0+1.18.2 will work around the problem.

Same exact problem.
Downgrading to even `fabric-api-0.53.3+1.18.2` works

**Comment by PoppetsMC** [comment]
> I had the same problem. fabric api 0.53.4+1.18.2 is the cause. fabric api 0.53.0+1.18.2 will work around the problem. 
I am having this issue too and just tried this and the game loaded but  Optifine didn't load, no shaders added :(  What version of Optifine? I am using the latest version
Thank you

**Comment by ChromaticIsobar** [comment]
To make it work, I am using
`fabric-api-0.53.3+1.18.2`
`optifabric-1.13.0`
`OptiFine_1.18.2_HD_U_H7`

**Comment by ChromaticIsobar** [comment]
> I am having this issue too and just tried this and the game loaded but  Optifine didn't load, no shaders added :(  What version of Optifine? I am using the latest version
> Thank you

To make it work, I am using
`fabric-api-0.53.3+1.18.2`
`optifabric-1.13.0`
`OptiFine_1.18.2_HD_U_H7`

**Comment by PoppetsMC** [comment]
@ChromaticIsobar  Version fabric 53.3 worked! Thank you so much & have a wonderful week <3

**Comment by osfanbuff63** [comment]
> > I am having this issue too and just tried this and the game loaded but  Optifine didn't load, no shaders added :(  What version of Optifine? I am using the latest version
> > Thank you
> 
> To make it work, I am using `fabric-api-0.53.3+1.18.2` `optifabric-1.13.0` `OptiFine_1.18.2_HD_U_H7`

Wait - H7 worked? That's unusual given we don't officially support H7 I believe (unless I missed a PR or commit somewhere lol)

**Comment by ChromaticIsobar** [comment]
Yes, I am using H7.
I believe I updated Optifine in the process of understanding which mod was causing the problem.
![2022-06-03_16 12 13-crop](https://user-images.githubusercontent.com/9718897/171871890-5f003cca-9f85-4f29-a3d4-543d9676d02b.png)

**Comment by Errorbot1122** [comment]
Question, why is this on the fork and not on the original OptiFabric 🤔 

*NVM, it been moved to here for some reson*

**Comment by Pixie-Kitty** [comment]
> > I had the same problem. fabric api 0.53.4+1.18.2 is the cause. fabric api 0.53.0+1.18.2 will work around the problem.
> 
> Same exact problem. Downgrading to even `fabric-api-0.53.3+1.18.2` works

Finally got around to playing again and this worked thank you so much for the solutions, saved me alot more stress 🙏
